janusgraph-dist/src/assembly/static/bin/gremlin-server.sh

```diff
@@ -1,45 +1,118 @@
+
+[[ -n "$DEBUG" ]] && set -x
 
-# ${BASH_SOURCE[0]} is the path to this file
 SOURCE="${BASH_SOURCE[0]}"
-# Set $BIN to the absolute, symlinkless path to $SOURCE's parent
-while [ -h "$SOURCE" ]; do
-    BIN="$( cd -P "$( dirname "$SOURCE" )" && pwd )"
+
+# Set JANUSGRAPH_BIN and JANUSGRAPH_HOME if they are unset
+if [[ -z "$JANUSGRAPH_HOME" ]] && [[ -z "$JANUSGRAPH_BIN" ]]; then
+  while [[ -h "$SOURCE" ]]; do
+    JANUSGRAPH_BIN="$( cd -P "$( dirname "$SOURCE" )" && pwd )"
     SOURCE="$(readlink "$SOURCE")"
     [[ $SOURCE != /* ]] && SOURCE="$BIN/$SOURCE"
-done
-BIN="$( cd -P "$( dirname "$SOURCE" )" && pwd )"
-# Set $CFG to $BIN/../conf/gremlin-server
-cd -P $BIN/../conf/gremlin-server
-CFG=$(pwd)
-# Set $LIB to $BIN/../lib
-cd -P $BIN/../lib
-LIB=$(pwd)
-# Set $LIB to $BIN/../ext
-cd -P $BIN/../ext
-EXT=$(pwd)
-# Initialize classpath to $CFG
-CP="$CFG"
-# Add the slf4j-log4j12 binding
-CP="$CP":$(find -L $LIB -name 'slf4j-log4j12*.jar' | sort | tr '\n' ':')
-# Add the jars in $BIN/../lib that start with "janusgraph"
-CP="$CP":$(find -L $LIB -name 'janusgraph*.jar' | sort | tr '\n' ':')
-# Add the remaining jars in $BIN/../lib.
-CP="$CP":$(find -L $LIB -name '*.jar' \
-                \! -name 'janusgraph*' \
-                \! -name 'slf4j-log4j12*.jar' | sort | tr '\n' ':')
-# Add the jars in $BIN/../ext (at any subdirectory depth)
-CP="$CP":$(find -L $EXT -name '*.jar' | sort | tr '\n' ':')
+  done
+  JANUSGRAPH_BIN="$( cd -P "$( dirname "$SOURCE" )" && pwd )"
+  echo"Hello"
+  echo $JANUSGRAPH_BIN
+  cd -P "$( dirname "$SOURCE" )" || exit 1
+  cd $JANUSGRAPH_BIN/..
+  JANUSGRAPH_HOME="$(pwd)"
+elif [[ -z "$JANUSGRAPH_HOME" ]]; then
+  cd $JANUSGRAPH_BIN/..
+  JANUSGRAPH_HOME="$(pwd)"
+elif [[ -z "$JANUSGRAPH_BIN" ]]; then
+  JANUSGRAPH_BIN="${JANUSGRAPH_HOME}/bin"
+fi
+
+# Set $JANUSGRAPH_CFG to $JANUSGRAPH_HOME/conf/gremlin-server if unset
+if [[ -z "$JANUSGRAPH_CFG" ]]; then
+  cd -P ${JANUSGRAPH_HOME}/conf/gremlin-server
+  JANUSGRAPH_CFG=$(pwd)
+fi
+# Set $JANUSGRAPH_LIB to $JANUSGRAPH_HOME/lib if unset
+if [[ -z "$JANUSGRAPH_LIB" ]]; then
+  cd -P $JANUSGRAPH_HOME/lib
+  JANUSGRAPH_LIB=$(pwd)
+fi
+# Set $JANUSGRAPH_EXT to $JANUSGRAPH_HOME/ext if unset
+if [[ -z "$JANUSGRAPH_EXT" ]]; then
+  cd -P $JANUSGRAPH_HOME/ext
+  JANUSGRAPH_EXT=$(pwd)
+fi
+
+if [[ -z "$JANUSGRAPH_CP" ]];then
+  # Initialize classpath to $JANUSGRAPH_CFG
+  JANUSGRAPH_CP="${JANUSGRAPH_CFG}"
+  # Add the slf4j-log4j12 binding
+  JANUSGRAPH_CP="$JANUSGRAPH_CP":$(find -L $JANUSGRAPH_LIB -name 'slf4j-log4j12*.jar' | sort | tr '\n' ':')
+  # Add the jars in $JANUSGRAPH_HOME/lib that start with "janusgraph"
+  JANUSGRAPH_CP="$JANUSGRAPH_CP":$(find -L $JANUSGRAPH_LIB -name 'janusgraph*.jar' | sort | tr '\n' ':')
+  # Add the remaining jars in $JANUSGRAPH_HOME/lib.
+  JANUSGRAPH_CP="$JANUSGRAPH_CP":$(find -L $JANUSGRAPH_LIB -name '*.jar' \
+                  \! -name 'janusgraph*' \
+                  \! -name 'slf4j-log4j12*.jar' | sort | tr '\n' ':')
```

Comment:
It's actually saying to not add those last 2 lines because of the `!`. I double checked in the man page to be safe.
```
find / \! -name "*.c" -print
             Print out a list of all the files whose names do not end in .c.
```